qiskit/pulse/commands/sample_pulse.py

```diff
@@ -15,35 +15,21 @@
 class SamplePulse(PulseCommand):
     """Container for functional pulse."""
 
-    def __init__(self, duration, samples, name):
+    def __init__(self, name, samples):
```

Comment:
The name should not be a required argument. A default name should be assigned on assembly time if otherwise not present. This way `samples` are the only required argument.

Reply:
Leaving this as `None` may cause a problem in the result of `__eq__` . A unique name is automatically assigned by object id in the constructor.

Reply:
A `SamplePulse` should be equal if its samples are equal. The `name` should only be an optional identifier for assembly in a Qobj